components/aero_sensors/HW/HW_i2c.c

```diff
@@ -0,0 +1,260 @@
+        if (HAL_DMA_Init(&hdma_i2c1_rx) != HAL_OK)
+        {
+            Error_Handler();
+        }
+
+        __HAL_LINKDMA(hi2c, hdmarx, hdma_i2c1_rx);
+
+        /**
+         * I2C1 Tx DMA configured on DMA Channel 3
+         */
+        hdma_i2c1_tx.Instance                 = DMA1_Channel6;
+        hdma_i2c1_tx.Init.Direction           = DMA_MEMORY_TO_PERIPH;
+        hdma_i2c1_tx.Init.PeriphInc           = DMA_PINC_DISABLE;
+        hdma_i2c1_tx.Init.MemInc              = DMA_MINC_ENABLE;
+        hdma_i2c1_tx.Init.PeriphDataAlignment = DMA_PDATAALIGN_BYTE;
+        hdma_i2c1_tx.Init.MemDataAlignment    = DMA_MDATAALIGN_BYTE;
+        hdma_i2c1_tx.Init.Mode                = DMA_NORMAL;
+        hdma_i2c1_tx.Init.Priority            = DMA_PRIORITY_MEDIUM;
+        if (HAL_DMA_Init(&hdma_i2c1_tx) != HAL_OK)
+        {
+            Error_Handler();
+        }
+
+        __HAL_LINKDMA(hi2c, hdmatx, hdma_i2c1_tx);
+
+    } else if (hi2c->Instance == I2C2)
+    {
+        /**
+         * I2C1 Rx DMA configured on DMA Channel 2
+         */
+        hdma_i2c2_rx.Instance                 = DMA1_Channel5;
+        hdma_i2c2_rx.Init.Direction           = DMA_PERIPH_TO_MEMORY;
+        hdma_i2c2_rx.Init.PeriphInc           = DMA_PINC_DISABLE;
+        hdma_i2c2_rx.Init.MemInc              = DMA_MINC_ENABLE;
+        hdma_i2c2_rx.Init.PeriphDataAlignment = DMA_PDATAALIGN_BYTE;
+        hdma_i2c2_rx.Init.MemDataAlignment    = DMA_MDATAALIGN_BYTE;
+        hdma_i2c2_rx.Init.Mode                = DMA_NORMAL;
+        hdma_i2c2_rx.Init.Priority            = DMA_PRIORITY_MEDIUM;
+        if (HAL_DMA_Init(&hdma_i2c2_rx) != HAL_OK)
+        {
+            Error_Handler();
+        }
+
+        __HAL_LINKDMA(hi2c, hdmarx, hdma_i2c2_rx);
+
+        /**
+         * I2C1 Tx DMA configured on DMA Channel 3
+         */
+        hdma_i2c2_tx.Instance                 = DMA1_Channel4;
+        hdma_i2c2_tx.Init.Direction           = DMA_MEMORY_TO_PERIPH;
+        hdma_i2c2_tx.Init.PeriphInc           = DMA_PINC_DISABLE;
+        hdma_i2c2_tx.Init.MemInc              = DMA_MINC_ENABLE;
+        hdma_i2c2_tx.Init.PeriphDataAlignment = DMA_PDATAALIGN_BYTE;
+        hdma_i2c2_tx.Init.MemDataAlignment    = DMA_MDATAALIGN_BYTE;
+        hdma_i2c2_tx.Init.Mode                = DMA_NORMAL;
+        hdma_i2c2_tx.Init.Priority            = DMA_PRIORITY_MEDIUM;
+        if (HAL_DMA_Init(&hdma_i2c2_tx) != HAL_OK)
+        {
+            Error_Handler();
+        }
+
+        __HAL_LINKDMA(hi2c, hdmatx, hdma_i2c2_tx);
+
+    }
+}
+
+/**
+ * @brief  Blocking mode write to I2C device
+ *
+ * @param dev I2C handle and address of device
+ * @param pData Data buffer
+ * @param Size Size (in bytes) to be transmitted
+ * @param Timeout Maximum timeout
+ */
+void HW_I2C_Master_Write(HW_I2C_Device_S *dev, uint8_t* pData, uint16_t Size, uint32_t Timeout)
+{
+    HAL_I2C_Master_Transmit(dev->handle, dev->addr, pData, Size, Timeout);
+}
```

Comment:
if all you're doing is passing the parameters to another function, these should all just be macros.